firewalls.tf

```diff
@@ -29,3 +29,20 @@ resource "google_compute_firewall" "bastion_deny_all_ingress" {
 
   target_service_accounts = [google_service_account.bastion[0].email]
 }
+
+resource "google_compute_firewall" "gke_knative_serving_webhook_allow" {
+  count       = var.enable_knative ? 1 : 0
+  name        = "${var.deployment_id}-knative-webhook-allow-ingress"
+  network     = google_compute_network.core.self_link
+  description = "Allow GKE master to communicate ti knative webhook"
+  priority    = 1000
+  direction   = "INGRESS"
+
+  allow {
+    protocol = "tcp"
+    ports    = ["8443"]
+  }
+
+  source_ranges = [google_container_cluster.primary.private_cluster_config.0.master_ipv4_cidr_block]
+  target_tags   = local.network_tags
```

Comment:
Please try replacing this with a compact() join of the two node pools’ tags, an attribute available in the terraform node_config.

Reply:
> Tags are not on firewalls, they are on targets - in this case GCP instances. Please check to see if we can use existing node pools by referencing their tags. The firewall will be dependent on both node pools.

Did you check the link ? (https://cloud.google.com/kubernetes-engine/docs/how-to/private-clusters#view_firewall_rules). Tags are on targets and are defined on `target_tag` in firewall rule. To get the tag created by Google/GKE is what is described in that link.

Reply:
> Please try replacing this with a compact() join of the two node pools’ tags, an attribute available in the terraform node_config.

Unfortunately, atleast I was not able to get that from there. Maybe I am missing something .

Reply:
The node_config.tags in terraform state is empty and it is not showing the tag that GKE creates.

![image](https://user-images.githubusercontent.com/8811558/68392974-e21a0400-0162-11ea-967f-3254ea887396.png)

Reply:
I prefer if we can find some way to get the existing (GKE) tags dynamically because I want to avoid recreating the node pools. Regarding if I read the link, yes I did. I also reviewed all the available terraform resources' and data sources' attributes to double-check, and also some related issues on github. I do understand the issue. I feel this is perhaps asking too much of you on this review, so I will give it a shot on the dev cluster now.

Reply:
I will try using the instance_group_urls attribute described here https://github.com/terraform-providers/terraform-provider-google/issues/2373#issuecomment-434787065 in combination with a instance group data source.

Reply:
I completely agree that if there is another way, we should definitely use that instead of haveing to re-create the node pool

Reply:
And yes I did think of https://www.terraform.io/docs/providers/google/d/google_compute_instance_group.html but it didn't show the attribute tags (or network_tags), hence didn't try it further but yeah if we can get that to work it would be ideal.

Thanks @sjmiller609

Reply:
One other workaround I was thinking of is running the gcloud command in a `null_resource` but it would also come with a caveat of having gcloud and properly configured with auth:

```
gcloud compute firewall-rules list \
    --filter 'name~^gke-[CLUSTER_NAME]' \
    --format 'table(
        name,
        network,
        direction,
        sourceRanges.list():label=SRC_RANGES,
        allowed[].map().firewall_rule().list():label=ALLOW,
        targetTags.list():label=TARGET_TAGS
    )'
```